mlx/traceability.py

```diff
@@ -120,20 +120,39 @@ def perform_replacement(self, app, collection):
         item_info = collection.get_item(self['reftarget'])
         if item_info:
             if not self.has_warned_about_undefined(item_info):
-                try:
-                    new_node = make_refnode(app.builder,
-                                            self['document'],
-                                            item_info.docname,
-                                            item_info.node['refid'],
-                                            self[0].deepcopy(),
-                                            self['reftarget'])
-                except NoUri:
-                    # ignore if no URI can be determined, e.g. for LaTeX output :(
-                    pass
+                notification_item_id = app.config.traceability_notifications.get('undefined-reference')
+                node = self._try_make_refnode(app, item_info.docname, item_info.node['refid'])
+                if node is None and notification_item_id:
+                    notification_item = app.env.traceability_collection.get_item(notification_item_id)
+                    if notification_item:
+                        node = self._try_make_refnode(app, notification_item.docname, notification_item_id)
```

Comment:
node can be none here and we wont trigger the warning. I think we should trigger the warnings here.

Reply:
Right now the warning(s) will be triggered later on when the dummy reference node(s) `ITEM_NOT_FOUND` cannot be resolved. This behavior of reporting warnings is not new and has not changed by this MR. However, I think it could be nice to have a custom warning for this specific case, i.e. when the configuration for a notification node exists, but the node itself does not.  Just know that each undefined node would trigger two warnings instead of one if we decide to add an additional, custom warning. Do you want me to implement this?

Reply:
I think this is a fair tradeoff if you wanted to hide an undefined reference but made a typo then this seems acceptable. I will let @dryodon  to decide as it is his feature request.

Reply:
Double warning is ok, let's implement the custom warning.

Reply:
Implemented by 6776b11bf81bf6527d448f1a6c67a09f4c235ba5 and d2247994c4591c9ddc2f4318f35b569223fcdabd